onnxruntime/core/session/abi_session_options_impl.h

```diff
@@ -14,6 +14,7 @@ struct OrtSessionOptions {
   onnxruntime::SessionOptions value;
   std::vector<OrtCustomOpDomain*> custom_op_domains_;
   std::vector<std::shared_ptr<onnxruntime::IExecutionProviderFactory>> provider_factories;
+  std::vector<std::shared_ptr<void>> custom_op_lib_handles_;
```

Comment:
>std::vector<std::shared_ptr<void>> custom_op_lib_handles_; [](http://example.com/codeflow?start=2&length=58)

I think wrapping the whole vector into shared_ptr with raw handles inside would be great since we are copying the instance of the options and we do not plan to share handles between unrelated instances of SessionOptions. Also if an unrelated options instance loads the same lib, then it is not an issue as the OS would still refcount them.
Also the whole vector becomes optional.

Reply:
Created separate PR for this: https://github.com/microsoft/onnxruntime/pull/14099